pkg/services/dns/dns.go

```diff
@@ -113,6 +146,41 @@ func (h *dnsHandler) addAnswers(m *dns.Msg) {
 					Ns: ns.Host,
 				})
 			}
+		case dns.TypeSRV:
+			_, records, err := resolver.LookupSRV(context.TODO(), "", "", q.Name)
+			if err != nil {
+				m.Rcode = dns.RcodeNameError
+				return
+			}
+			for _, srv := range records {
+				m.Answer = append(m.Answer, &dns.SRV{
+					Hdr: dns.RR_Header{
+						Name:   q.Name,
+						Rrtype: dns.TypeSRV,
+						Class:  dns.ClassINET,
+						Ttl:    0,
+					},
+					Port:     srv.Port,
+					Priority: srv.Priority,
+					Target:   srv.Target,
+					Weight:   srv.Weight,
+				})
+			}
+		case dns.TypeTXT:
+			records, err := resolver.LookupTXT(context.TODO(), q.Name)
+			if err != nil {
+				m.Rcode = dns.RcodeNameError
+				return
+			}
+			m.Answer = append(m.Answer, &dns.TXT{
+				Hdr: dns.RR_Header{
+					Name:   q.Name,
+					Rrtype: dns.TypeTXT,
+					Class:  dns.ClassINET,
+					Ttl:    0,
+				},
+				Txt: records,
+			})
```

Comment:
So we are implementing all the types except `AAAA` one?

Reply:
I implemented the types which have a corresponding method in https://pkg.go.dev/net#Resolver
I did not check if there are other DNS types not available through these methods
AAAA is for ipv6 records, gvisor-tap-vsock has no ipv6 support at the moment, so I don't know if they work or not.

Reply:
We had discussed this already and decided that IPv6 needs a lot more work. Adding AAAA record support would give the wrong idea and possibly even result in failures due to other missing functionality.

Reply:
AAAA support is part of https://github.com/containers/gvisor-tap-vsock/pull/59/commits ( https://github.com/containers/gvisor-tap-vsock/pull/59/commits/964d1136e8fec5c023cefb18bf73346a2dafaaa8 )